packages/deployment/app/cloudformation.template.yml

```diff
@@ -1018,12 +1018,13 @@ Resources:
             Value: {{{this}}}
       {{/if}}
     {{/each}}
-      {{# if ../ecs.docker}}
+    {{# ifEquals ../ecs.docker.registry "dockerhub"}}
         Image: {{image}}
-      {{else}}
+    {{/ifEquals}}
+    {{# ifEquals ../ecs.docker.registry "ecr"}}
         Image:
           Fn::Sub: ${AWS::AccountId}.dkr.ecr.${AWS::Region}.amazonaws.com/{{image}}
```

Comment:
This may break previous configurations that were using the ECR registry, as the template was ignoring the default 'registry' value, meaning you could omit the 'docker' object, have an image  specified and have it use the correct ecr image values.   I realize "that's the bug", but.....   Is this being treated as an inadvertent breaking change for the next release, or am I being too paranoid? 

If yes, we may need a CHANGELOG entry/user notification of some sort.

Reply:
I don't think you're being too paranoid - this is something that probably would break anything that _currently_ works with ECR registries.

I would prefer that we be able to support both reference schemes (full URI and short-name), but I didn't see a way to support that with `kes` as-is, and didn't look into writing new functionality/overrides.

Thinking about this a little more, I don't see how my changes here are any better than what we currently have except these changes will force current users to change their service.image values...

Reply:
Actually looking at this a bit more closely, for it to fail that way, a user would have to undefine docker as the default is to have it set in deployment/app/config.yml.       I'm far less convinced after re-reading this that users are undefining that value given the docs don't suggest that's the right way to do it.

Reply:
The issue with the value for image itself, however, remains.